hudi-client/hudi-spark-client/src/test/java/org/apache/hudi/client/functional/TestHoodieBackedTableMetadata.java

```diff
@@ -285,6 +297,112 @@ public void testMetadataRecordKeyExcludeFromPayload(final HoodieTableType tableT
     validateMetadata(testTable);
   }
 
+  /**
+   * This tests the case where the two clean actions delete the same file and commit
+   * to the metadata table. The metadata table should not contain the deleted file afterwards.
+   * A new cleaner plan may contain the same file to delete if the previous cleaner
+   * plan has not been successfully executed before the new one is scheduled.
+   */
+  @ParameterizedTest
+  @EnumSource(HoodieTableType.class)
```

Comment:
Do we need this functional test? 
I see we have added tests against HoodieMetadataPayload. 
trying to avoid writing end to end functional tests if we can test the fix/functionality using UTs.

Reply:
esply w/ MDT, we have been adding lot of functional tests in general. what do you think

Reply:
This one runs quickly as the data files are dummy.  Given the criticality of MDT logic, I think it's still worthwhile adding a functional tests here to make sure assumption is intact.

Reply:
Since we're improving MDT in 1.x releases, these tests serve as extra guards around the correctness.